draft-jennings-perc-double-01.md

```diff
@@ -96,127 +101,176 @@ Terms:
+  defined outside of this document that the MDD is likely to change.
+{Note: I don't like this step, because it doesn't really save
+any work for the MDD.  The MDD has to look for the field and get prepared
+to insert it or change it.  By the time it does that, there is really
+very little work left.  And I don't like it being mandatory, either, because
+the only logical mandatory requirement would be the 3-octet variety, thus
+making every packet 3 octets longer.  I prefer we just leave this to the MDD.}
 
-* Apply the SRTP cryptographic transform with the outer parameters (outer
-  transform)
+* Apply the outer cryptographic transform to the RTP packet
 
 
 ## Modifying a Packet
 
 In order to modify a packet, the MDD undoes the outer transform, modifies the
-packet, updates the OHB with any new modifications, and re-applies the outer
-tranform.
-
-* Apply the (outer) decryption transform to the packet
+packet, updates the OHB with any modifications not already present in the OHB,
+and re-applies the outer transform.
 
-* Separate the OHB from the (encrypted) original payload
+* Apply the outer decryption transform to the packet
 
 * Change any required parameters
 
-* If a changed parameter is not already in the OHB, add it with its original
-  value to the OHB. Note that in the case of cascaded MDDs, the first MDD may
-  have already added an OHB.
+* If a changed RTP header field is not already in the OHB, add it with its
+  original value to the OHB.  Note that in the case of cascaded MDDs, the
+  first MDD may have already added an OHB.
+{Note: if we allow the endpoint to add the OHB, we need to revise the
+above sentence to reflect the endpoint's contribution.}
 
 * If the MDD resets a parameter to its original value, it MAY drop it from the
-  OHB.
+  OHB as long as there are no other header extensions following the OHB.
 
 * The MDD MUST NOT delete any header extensions, but MAY add them.
 
     * If the MDD adds any header extensions, it must append them and it must
       maintain the order of the original headers in the {{RFC5285}} block.
     
-    * If the MDD appends headers, then it MUST add the value of the original
-      {{RFC5285}} length field to the OHB, or update it if it is already
-      there. The original {{RFC5285}} length is counted in words and stored in
-      the Ext Len field of the OHB.
-
-* Recombine the new OHB and the (encrypted) original payload
-
-* Apply the (outer) encryption transform to the packet
+    * If the MDD appends header extensions, then it MUST add the OHB header
+      extension (if not present) and add them following the OHB.  The OHB
+      serves as a demarcation point between original RTP header extensions
+      introduced by the endpoint and those introduced by an MDD.  If the
+      MDD did not make changes that would otherwise require an OHB, then
+      it SHOULD insert an OHB that merely replicated the unchanged RTP header
+      field values unchanged.
```

Comment:
Think about the case where a prior MDD added a OHB and some header extensions after the OHB. Now this MDD is going to add even some more header extensions. Clearly the OHB needs to come before the header extensions the first MDD added. I'm not sure we have the text right for this case.